x-pack/plugins/enterprise_search/public/applications/analytics/components/add_analytics_collections/add_analytics_collection_logic.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { AnalyticsCollection } from '../../../../../common/types/analytics';
+import { HttpError, Status } from '../../../../../common/types/api';
+
+import { isAlphaNumericOrUnderscore } from '../../../../../common/utils/is_alphanumeric_underscore';
+import { generateEncodedPath } from '../../../shared/encode_path_params';
+import { flashAPIErrors, flashSuccessToast } from '../../../shared/flash_messages';
+import { KibanaLogic } from '../../../shared/kibana';
+import { AddAnalyticsCollectionsAPILogic } from '../../api/add_analytics_collection/add_analytics_collection_api_logic';
+import { COLLECTION_VIEW_PATH } from '../../routes';
+
+export interface AddAnalyticsCollectionsActions {
+  apiError(error: HttpError): HttpError;
+  apiSuccess(collection: AnalyticsCollection): AnalyticsCollection;
+  createAnalyticsCollection(): void;
+  makeRequest: typeof AddAnalyticsCollectionsAPILogic.actions.makeRequest;
+  setInputError(error: string | boolean): { inputError: string | boolean };
+  setNameValue(name: string): { name: string };
+}
+
+interface AddAnalyticsCollectionValues {
+  canSubmit: boolean;
+  hasInputError: boolean;
+  inputError: string | boolean;
```

Comment:
question: Can we simplify this to being just a string? In what case would this be a boolean vs a string?

Reply:
Yeh, I originally had it as a string only but felt ugly when to clear the error was setting it to an empty string. I think ill flip back on this and the type could be strictly a string or a null.